modules/exploits/windows/local/docker_credential_wincred.rb

```diff
@@ -0,0 +1,107 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ManualRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+  include Post::Windows::Priv
+  include Post::Windows::Runas
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'          => 'Docker-Credential-Wincred.exe priviledge escalation',
+        'Description'   => %q{
+          This exploit leverages a vulnerability in docker desktop
+          community editions prior to 2.1.0.1 where an attacker can write
+          a payload to a lower-priviledged area to be executed
+          automatically by the docker user at login.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [
+          'Morgan Roman', # discovery
+          'bwatters-r7',  # metasploit module
+        ],
+        'Platform'       => ['win'],
+        'SessionTypes'   => ['meterpreter'],
+        'Targets'        => [[ 'Automatic', {} ]],
+        'DefaultTarget'  => 0,
+        'DefaultOptions' => {
+          'WfsDelay'     => 15
+        },
+        'DisclosureDate' => '2019-07-05',
+        'Notes'          =>
+        {
+          'SideEffects'    => [ ARTIFACTS_ON_DISK ]
```

Comment:
@adfoster-r7: What happened here with RuboCop?

Reply:
Hm, I got very different results when I ran Rubocop locally before suggesting to run it:

```
rubocop -a modules/exploits/windows/local/docker_credential_wincred.rb
```

Which resulted in a format of:

```ruby
  def initialize(info = {})
    super(
      update_info(
        info,
        'Name' => 'Docker-Credential-Wincred.exe priviledge  escallation',
        'Description' => %q{
          This exploit leverages a vulnerability in docker desktop
          community editions prior to 2.1.0.1 where an attacker can write
          a payload to a lower-priviledged area to be executed
          automatically by the docker user at login.
        },
        'License' => MSF_LICENSE,
        'Author' => [
          'Morgan Roman', # discovery
          'bwatters-r7', # metasploit module
        ],
        'Platform' => ['win'],
        'SessionTypes' => ['meterpreter'],
        'Targets' => [[ 'Automatic', {} ]],
        'DefaultTarget' => 0,
        'DefaultOptions' => {
          'WfsDelay' => 15
        },
        'DisclosureDate' => '2019-07-05',
        'Notes' =>
        {
          'SideEffects' => [ ARTIFACTS_ON_DISK ]
        },
        'References' => [
          ['CVE', '2019-15752'],
          ['URL', 'https://medium.com/@morgan.henry.roman/elevation-of-privilege-in-docker-for-windows-2fd8450b478e']
        ]
      )
    )
    register_options(
      [OptString.new('PROGRAMDATA', [true, 'Path to docker version-bin.', 'C:\\ProgramData\\DockerDesktop\\version-bin\\'])]
    )
  end
```

@bwatters-r7 I wonder if you have any custom Ruby formatters in your editor that changed the rocket syntax? 🤔

Reply:
Are we running rubocop with autocorrect, now?

Reply:
@bwatters-r7 That's the plan 🎉Rubocop's now been been configured to produce pretty sane output now by default, it required an extra custom rule or two to make that happen https://github.com/rapid7/metasploit-framework/pull/12990